### PR TITLE
Added support for enhanced enums in data/demos.dart

### LINKED ISSUE #680: Use enhanced enums in `data/demos.dart`
More details at https://medium.com/dartlang/dart-2-17-b216bfc80c5d

### PR DESCRIPTION
**Pull request by ahmadabbas02**
This is my first pull request and I'm a new contributor

My pull request should fix issue #680 where I overrode the toString() function and I had to change the minimum SDK in pubspec.yaml to 2.17 to support enhanced enums

Fixes #680

### CODE REVIEW

File: lib/data/demos.dart [hunk 1]
```diff
@@ -41,18 +41,22 @@ enum GalleryDemoCategory {
...
-  other,
+  other;
+
+  @override
+  String toString() {
+    return name.toUpperCase();
+  }
 }
 
 extension GalleryDemoExtension on GalleryDemoCategory {
```
Comment (guidezpl) [inline lib/data/demos.dart:52]: Hi, happy to have you contribute! 

Enhanced enums allow us to get rid of this extension, altogether. Can we move displayTitle inside the enum?

Comment (ahmadabbas02) [reply lib/data/demos.dart]: Hey, yes sure i fixed this in the new commit

File: pubspec.lock [hunk 2]
```diff
@@ -49,7 +49,7 @@ packages:
       name: async
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "2.9.0"
+    version: "2.8.2"
```
Comment (guidezpl) [inline pubspec.lock:52]: Please revert this

Comment (ahmadabbas02) [reply pubspec.lock]: done, 1 extra commit was created due to me having to pull from master cuz of conflicts

**Review by guidezpl — APPROVED** [review]

**Pull request merged**